### ISSUE TITLE
Input/file reader

### ISSUE THREAD

**Issue by aroberts89** [ucd-os-blue-s18/msl-clang-002#3]
### Requirements
* The input will be a file containing English words, one word per line. The words are not guaranteed to be lower-case for this assignment, but your counts need to be case-INsensitive. You need to read it in to sort them. The test file(s) will be in the same directory as your executable and will appear as the first argument when your program is run.
* Read words from the file. Don't read in the whole file and then process it. Read it in chunks using a buffer.

---

In other words, looks pretty similar to what we already did on the first one, so you should be able to _mostly_ do some copying from there. The case insensitive point should only matter later when we go to do tree insertion ( in issue #1 )
